UI/Components/ServerComponent.cs

```diff
@@ -172,199 +172,230 @@ private void ProcessMessage(String message, Connection clientConnection)
+                    }
+                    case "getfinaltime":
+                    case "getfinalsplittime":
+                    {
+                        var comparison = args.Length > 1 ? args[1] : State.CurrentComparison;
+                        var time = (State.CurrentPhase == TimerPhase.Ended)
+                            ? State.CurrentTime[State.CurrentTimingMethod]
+                            : State.Run.Last().Comparisons[comparison][State.CurrentTimingMethod];
+                        var response = SplitTimeFormatter.Format(time);
+                        clientConnection.SendMessage(response);
+                        break;
+                    }
+                    case "getbestpossibletime":
+                    case "getpredictedtime":
+                    {
+                        string comparison;
+                        if (command == "getbestpossibletime")
+                            comparison = LiveSplit.Model.Comparisons.BestSegmentsComparisonGenerator.ComparisonName;
+                        else
+                            comparison = args.Length > 1 ? args[1] : State.CurrentComparison;
+                        var prediction = PredictTime(State, comparison);
+                        var response = SplitTimeFormatter.Format(prediction);
+                        clientConnection.SendMessage(response);
+                        break;
+                    }
+                    case "gettimerphase":
+                    case "getcurrenttimerphase":
+                    {
+                        var response = State.CurrentPhase.ToString();
+                        clientConnection.SendMessage(response);
+                        break;
+                    }
+                    case "setcomparison":
+                    {
+                        State.CurrentComparison = args[1];
+                        break;
+                    }
+                    case "switchto":
+                    {
+                        State.CurrentTimingMethod = args[1] == "gametime" ? TimingMethod.GameTime : TimingMethod.RealTime;
+                        break;
+                    }
+                    case "setsplitname":
+                    case "setcurrentsplitname":
+                    {
+                        var index = State.CurrentSplitIndex;
+                        var title = args[1];
+
+                        if (command == "setsplitname")
+                        {
+                            var options = args[1].Split(new [] { ' ' }, 2);
+                            index = Convert.ToInt32(options[0]);
+                            title = options[1];
+                        }
+
+                        State.Run[index].Name = title;
+                        State.Run.HasChanged = true;
+                        break;
+                    }
+                    default:
+                    {
+                        // perhaps an error should be returned for an unrecognized message?
```

Comment:
We could send an "Unrecognized command" message back to the client.

Reply:
Will do. I actually think there should also be a message sent on errors so that clients don't just have to timeout (and can then better detect connection issues separately from errors). Could we standardize these two things? Could respond with `error "Error message"`, but I wasn't sure if there should be some kind of special identifier before "error", since technically "error" is a perfectly valid split name.

Reply:
That's a great point - maybe something like `[Error]: `?

Reply:
Yeah that seems fine enough to me. I noticed a couple bugs while working on this PR, so I'll add the Error response first so I have errors to reproduce with, and then fix the bugs. This will end up changing the behavior on error in any libraries that communicate with LS.Server though, so it may be best to consider it a breaking change for versioning purposes.